src/python/pants/engine/http_test.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from contextlib import contextmanager
+from http.server import BaseHTTPRequestHandler
+from typing import Dict, Iterator
+from urllib.parse import parse_qs, urlparse
+
+from pants.engine.http import HttpGetResponse, HttpRequester
+from pants.testutil.test_base import TestBase
+from pants.util.contextutil import http_server
+
+
+class HttpHandlerForTests(BaseHTTPRequestHandler):
+  response_text = b"A valid HTTP response!"
+  not_found_text = b"Custom 404 page"
+  server_failure_text = b"500 internal server error"
+
+  def do_GET(self):
+    self.send_headers()
+    if self.path == "/valid-url":
+      self.wfile.write(self.response_text)
+    elif self.path == "/deliberate500":
+      self.wfile.write(self.server_failure_text)
+    elif "test-headers" in self.path:
+      output = str(parse_qs(urlparse(self.path).query))
```

Comment:
It looks like `parse_qs` returns a `ParseResult`. Could we extract the information in a more structured way or `str()` is the best way to go? https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse